charts/kube-prometheus-stack/templates/prometheus/networkpolicy.yaml

```diff
@@ -24,7 +24,11 @@ spec:
     {{- toYaml .Values.prometheus.networkPolicy.podSelector | nindent 4 }}
     {{- else }}
     matchExpressions:
+      {{- if .Values.prometheus.agentMode }}
+      - {key: app.kubernetes.io/name, operator: In, values: [prometheus-agent]}
+      {{- else }}
       - {key: app.kubernetes.io/name, operator: In, values: [prometheus]}
-      - {key: prometheus, operator: In, values: [{{ template "kube-prometheus-stack.prometheus.crname" . }}]}
+      {{- end }}
+      - {key: operator.prometheus.io/name, operator: In, values: [{{ template "kube-prometheus-stack.prometheus.crname" . }}]}
```

Comment:
Hi @vholer, If I'm reading this right, this change goes against [this](https://github.com/prometheus-operator/prometheus-operator/blob/45a45f9737ae33b5d0eb4dddfbb6ee5444c3dfbd/pkg/prometheus/server/statefulset.go#L351) as you are changing the key `prometheus` to `operator.prometheus.io/name`.

Reply:
Yes, idea is to switch to different label, which is set both for Prometheus server and agent. This "new" label is set to same value `promName` in operator.

See:

https://github.com/prometheus-operator/prometheus-operator/blob/45a45f9737ae33b5d0eb4dddfbb6ee5444c3dfbd/pkg/prometheus/server/statefulset.go#L353

https://github.com/prometheus-operator/prometheus-operator/blob/45a45f9737ae33b5d0eb4dddfbb6ee5444c3dfbd/pkg/prometheus/statefulset.go#L62

Reply:
Got it! Didn't follow through to check the variable value.